frontend/src/features/seeds/api/searchPlants.ts

```diff
@@ -2,16 +2,30 @@ import { PlantsSummaryDto } from '@/bindings/definitions';
 import { baseApiUrl } from '@/config';
 import axios from 'axios';
 
-export const searchPlants = async (searchTerm: string): Promise<PlantsSummaryDto[]> => {
+export interface SearchResult {
+  dtos: PlantsSummaryDto[],
+  hasMore: boolean,
+}
+
+export const searchPlants = async (searchTerm: string, page: number): Promise<SearchResult> => {
   try {
+    // A page must always be defined in order for the search to work.
+    const pageString: string = page != undefined ? page.toString() : '1';
+
     const searchParams = new URLSearchParams();
     searchParams.append('search_term', searchTerm);
     searchParams.append('limit', '10');
+    searchParams.append('page', pageString);
 
     const response = await axios.get<PlantsSummaryDto[]>(
       `${baseApiUrl}/api/plants?${searchParams}`,
     );
-    return response.data;
+    
+    return {
+      dtos: response.data,
+      // If a page could not be filled completely, there will propably be no more data.
+      hasMore: response.data.length == 10,
```

Comment:
Yes, in the edge case you mentioned the backend would just return an empty page for the last request.  The frontend would then pass said empty page to the select menu component with the hasMore flag set to false.
So in short this would just result in a single unnecessary request.    

Loading an additional page on each request would require some state management to avoid doubling the network traffic. 

So in short I chose to do it this way to keep the code nice and simple, while avoiding additional  requests.

Reply:
> Loading an additional page on each request would require some state management to avoid doubling the network traffic.

I did not mean to load an additional page. Just one more item per request in the backend database query.
```rs
...
.limit(limit.into() + 1)
...

// something like this
let has_more = query_result.len() == limit + 1
```
but it does not matter anyway. The UX feels good.

Reply:
Ok sorry, I encountered some brain-lag there :sweat_smile:.
Forget what I said, this is an absolutely great suggestion.